src/components/CourseRunRedirect/index.jsx

```diff
@@ -0,0 +1,33 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Redirect } from 'react-router';
+
+import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';
+
+class CourseRunRedirectComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { path: null };
+  }
+
+  async componentDidMount() {
+    try {
+      const response = await DiscoveryDataApiService
+        .fetchCourseRuns({ keys: this.props.courseRunKey });
+      const uuid = response.data.results[0].course_uuid;
+      this.setState({ path: `/courses/${uuid}` }); // eslint-disable-line react/no-did-mount-set-state
+    } catch (e) {
+      this.setState({ path: '/' }); // eslint-disable-line react/no-did-mount-set-state
+    }
+  }
+
+  render() {
+    return this.state.path ? <Redirect to={this.state.path} /> : null;
```

Comment:
This probably shouldn't be rendering null and instead render the `LoadingSpinner` component so it is consistent with actually loading a course.